app/helpers/address_helper.rb

```diff
@@ -9,7 +9,7 @@ def postcode_lookup(postcode)
     response = Faraday.get(postcode_url)
     details = JSON.parse(response.body)
 
-    raise AddressLookupError, details["error"] unless response.status == 200
+    raise AddressLookupError, response.status unless response.status == 200
 
     addresses = details["results"]
```

Comment:
`addresses` can be `nil`, so calling `addresses.map` below will fail. You could do `details.fetch("results", [])` and also check if `"totalresults"` is 0.

Reply:
Heya, can you explain this a bit more. As far as we knew this was taking all of the addresses for a given postcode. Does this mean that there are postcodes without addresses? Do you have and example? 

We were expecting this to either return an empty array or to return a 400 before it got to this point.

Reply:
Updated to use response.status.  Yes, there are postcodes without addresses (e.g. SW1A3AA) that we have accounted for.